workflow-manager/src/domain/cosmos_cw.rs

```diff
@@ -848,4 +1015,44 @@ impl CosmosCosmwasmConnector {
             ))?
             .data_hash)
     }
+
+    pub async fn get_code_id_of_addr(&mut self, addr: String) -> Result<u64, CosmosCosmwasmError> {
+        // find the code id of the address
+        let code_id_req = QueryContractInfoRequest {
+            address: addr.clone(),
+        };
+
+        // Query for the code id of the address
+        Ok(self
+            .wallet
+            .client
+            .clients
+            .wasm
+            .contract_info(code_id_req)
+            .await
+            .context(format!("Failed to query address info: {}", addr))
+            .map_err(CosmosCosmwasmError::Error)?
+            .into_inner()
+            .contract_info
+            .context(format!("No contract info found: {}", addr))
+            .map_err(CosmosCosmwasmError::Error)?
+            .code_id)
+    }
+
+    pub async fn get_contract_name_by_code_id(
+        &mut self,
+        addr: String,
+    ) -> Result<String, CosmosCosmwasmError> {
+        let code_id = self.get_code_id_of_addr(addr.clone()).await?;
+
```

Comment:
nit: was a bit confused here. Name is get_contract_name_by_code_id but you are actually passing an address as argument and getting the code_id inside

Reply:
Yup, started with a single function and separated them.
Fixed.